cli/lib/install.go

```diff
@@ -36,69 +36,99 @@ func initInstallCommand() *cobra.Command {
-				os.Exit(errorcodes.ErrGeneric)
+		if len(args) == 0 {
+			cmd.Help()
+			os.Exit(errorcodes.ErrGeneric)
+		}
+		libRefs, err := ParseLibraryReferenceArgsAndAdjustCase(instance, args)
+		if err != nil {
+			feedback.Errorf("Arguments error: %v", err)
+			os.Exit(errorcodes.ErrBadArgument)
+		}
+
+		toInstall := map[string]*rpc.LibraryDependencyStatus{}
+		if installFlags.noDeps {
+			for _, libRef := range libRefs {
+				toInstall[libRef.Name] = &rpc.LibraryDependencyStatus{
+					Name:            libRef.Name,
+					VersionRequired: libRef.Version,
+				}
 			}
-			for _, dep := range depsResp.GetDependencies() {
-				feedback.Printf("%s depends on %s@%s", libRef, dep.GetName(), dep.GetVersionRequired())
-				if existingDep, has := toInstall[dep.GetName()]; has {
-					if existingDep.GetVersionRequired() != dep.GetVersionRequired() {
-						// TODO: make a better error
-						feedback.Errorf("The library %s is required in two different versions: %s and %s",
-							dep.GetName(), dep.GetVersionRequired(), existingDep.GetVersionRequired())
-						os.Exit(errorcodes.ErrGeneric)
+		} else {
+			for _, libRef := range libRefs {
+				depsResp, err := lib.LibraryResolveDependencies(context.Background(), &rpc.LibraryResolveDependenciesReq{
+					Instance: instance,
+					Name:     libRef.Name,
+					Version:  libRef.Version,
+				})
+				if err != nil {
+					feedback.Errorf("Error resolving dependencies for %s: %s", libRef, err)
+					os.Exit(errorcodes.ErrGeneric)
+				}
+				for _, dep := range depsResp.GetDependencies() {
+					feedback.Printf("%s depends on %s@%s", libRef, dep.GetName(), dep.GetVersionRequired())
+					if existingDep, has := toInstall[dep.GetName()]; has {
+						if existingDep.GetVersionRequired() != dep.GetVersionRequired() {
+							// TODO: make a better error
+							feedback.Errorf("The library %s is required in two different versions: %s and %s",
+								dep.GetName(), dep.GetVersionRequired(), existingDep.GetVersionRequired())
+							os.Exit(errorcodes.ErrGeneric)
+						}
 					}
+					toInstall[dep.GetName()] = dep
 				}
-				toInstall[dep.GetName()] = dep
 			}
 		}
-	}
 
-	for _, library := range toInstall {
-		libraryInstallReq := &rpc.LibraryInstallReq{
-			Instance: instance,
-			Name:     library.Name,
-			Version:  library.VersionRequired,
-		}
-		err := lib.LibraryInstall(context.Background(), libraryInstallReq, output.ProgressBar(), output.TaskProgress())
-		if err != nil {
-			feedback.Errorf("Error installing %s: %v", library, err)
-			os.Exit(errorcodes.ErrGeneric)
+		for _, library := range toInstall {
+			libraryInstallReq := &rpc.LibraryInstallReq{
+				Instance: instance,
+				Name:     library.Name,
+				Version:  library.VersionRequired,
+			}
+			err := lib.LibraryInstall(context.Background(), libraryInstallReq, output.ProgressBar(), output.TaskProgress())
+			if err != nil {
+				feedback.Errorf("Error installing %s: %v", library, err)
+				os.Exit(errorcodes.ErrGeneric)
+			}
```

Comment:
I'm not sure setting the minimum number of args to `0` is the best approach here.
Let's try this thing, change the minimum number of args back to `1`, the flags `git-url` and `zip-path` to boolean flags. Doing so we can call the CLI like so: 
```
arduino-cli lib install <my_lib_name>
arduino-cli lib install --git-url <my_git_url>
arduino-cli lib install --zip-path <my_zip_file>
```
You'll have to change a bit the logic in this file but I think it's a good improvement and will make the command clearer to use.